adapters/Unity/UnityAdapter/GADMAdapterUnity.m

```diff
@@ -86,61 +104,36 @@ - (instancetype)initWithGADMAdNetworkConnector:(id<GADMAdNetworkConnector>)conne
-  _gameID = [[[strongConnector credentials] objectForKey:kGADMAdapterUnityGameID] copy];
-  _placementID = [[[strongConnector credentials] objectForKey:kGADMAdapterUnityPlacementID] copy];
-  if (!_gameID || !_placementID) {
-    NSError *error = GADMAdapterUnityErrorWithCodeAndDescription(
-        GADMAdapterUnityErrorInvalidServerParameters, @"Game ID and Placement ID cannot be nil.");
-    [strongConnector adapter:self didFailAd:error];
-    return;
-  }
-
-  [_metaData setCategory:@"mediation_adapter"];
-  [_metaData set:_uuid value:@"load-interstitial"];
-  [_metaData set:_uuid value:_placementID];
-  [_metaData commit];
-
-  [[GADMAdapterUnitySingleton sharedInstance] requestInterstitialAdWithDelegate:self];
+    id<GADMAdNetworkConnector> strongConnector = _networkConnector;
+    _gameID = [[[strongConnector credentials] objectForKey:kGADMAdapterUnityGameID] copy];
+    _interstitialAd = [[GADMUnityInterstitialAd alloc] initWithGADMAdNetworkConnector:strongConnector adapter:self];
+    [self initializeWithGameID:_gameID];
+    [_interstitialAd getInterstitial];
 }
 
 - (void)presentInterstitialFromRootViewController:(UIViewController *)rootViewController {
-  [_networkConnector adapterWillPresentInterstitial:self];
-
-  if (![UnityAds isReady:_placementID]) {
-    [_networkConnector adapterDidDismissInterstitial:self];
-    [_metaData setCategory:@"mediation_adapter"];
-    [_metaData set:_uuid value:@"fail-to-show-interstitial"];
-    [_metaData set:_uuid value:_placementID];
-    [_metaData commit];
-    return;
-  }
-
-  [_metaData setCategory:@"mediation_adapter"];
-  [_metaData set:_uuid value:@"show-interstitial"];
-  [_metaData set:_uuid value:_placementID];
-  [_metaData commit];
-  [[GADMAdapterUnitySingleton sharedInstance]
-      presentInterstitialAdForViewController:rootViewController
-                                    delegate:self];
+    id<GADMAdNetworkConnector> strongConnector = _networkConnector;
+    [strongConnector adapterWillPresentInterstitial:self];
+    [_interstitialAd presentInterstitialFromRootViewController:rootViewController];
 }
 
 #pragma mark Banner Methods
 
 - (void)getBannerWithSize:(GADAdSize)adSize {
-  id<GADMAdNetworkConnector> strongConnector = _networkConnector;
-  _gameID = [strongConnector.credentials[kGADMAdapterUnityGameID] copy];
-  _placementID = [strongConnector.credentials[kGADMAdapterUnityPlacementID] copy];
-  if (!_gameID || !_placementID) {
-    NSError *error = GADMAdapterUnityErrorWithCodeAndDescription(
-        GADMAdapterUnityErrorInvalidServerParameters, @"Game ID and Placement ID cannot be nil.");
-    [strongConnector adapter:self didFailAd:error];
-    return;
-  }
-
-  _bannerAd = [[GADMAdapterUnityBannerAd alloc] initWithGADMAdNetworkConnector:strongConnector
-                                                                       adapter:self];
-  [_bannerAd loadBannerWithSize:adSize];
+    id<GADMAdNetworkConnector> strongConnector = _networkConnector;
```

Comment:
do we need to check if `strongConnector` is not `nil`?

Reply:
Yes, we should. On it.

Reply:
why do we call this variable strong connector?

Reply:
That's because we are declaring a strong reference to the connector object and use it to get gameID and placementID. This is a good practice to avoid referring to nil values.